python/ray/train/callbacks/results_preprocessors/aggregate/aggregate_utils.py

```diff
@@ -0,0 +1,86 @@
+import logging
+from typing import Dict, List, Tuple, Union
+
+import numpy as np
+
+VALID_AGGREGATE_TYPES: Tuple[type] = (
+    int,
+    float,
+    np.float32,
+    np.float64,
+    np.int32,
+    np.int64,
+)
+
+logger = logging.getLogger(__name__)
+
+
+def _get_metrics_from_results(
+    key: str, results: List[Dict]
+) -> List[Union[VALID_AGGREGATE_TYPES]]:
+    """Return metrics values in the results list from all workers.
+
+    Args:
+        key (str): A key string. If it doesn't exist in results,
+            i.e. it is not reported, the None will be returned.
+        results (List[Dict]): The results list returned from workers.
+
+    Returns:
+        A list of valid key values from each worker, if key exists.
+        Otherwise, None.
+    """
+    reported_metrics = {key for result in results for key in result.keys()}
+    values = [result.get(key, np.nan) for result in results]
+    warning_message = None
+    if key not in reported_metrics:
```

Comment:
I don't think this is doing what you intend it to do? Do you want to return None if the key is not reported by _any_ single worker? Or if it's not included in _all_ workers?